triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Runner.scala

```diff
@@ -849,23 +860,72 @@ private[lf] class Runner(
       acs: Seq[CreatedEvent],
       offset: LedgerOffset,
       msgFlow: Graph[FlowShape[TriggerMsg, TriggerMsg], T] = Flow[TriggerMsg],
-      name: String = "",
   )(implicit
       materializer: Materializer,
       executionContext: ExecutionContext,
   ): (T, Future[SValue]) = {
     val source =
       msgSource(client, offset, trigger.heartbeat, parties, transactionFilter)
+
     Flow
       .fromGraph(msgFlow)
-      .viaMat(getTriggerEvaluator(name, acs))(Keep.both)
+      .viaMat(getTriggerEvaluator(acs))(Keep.both)
       .via(submitOrFail)
       .join(source)
       .run()
   }
 }
 
-object Runner extends StrictLogging {
+object Runner {
+
+  import Implicits._
+
+  private[trigger] val logger = ContextualizedLogger.get(getClass)
+
+  def apply(
+      compiledPackages: CompiledPackages,
+      trigger: Trigger,
+      triggerConfig: TriggerRunnerConfig,
+      client: LedgerClient,
+      timeProviderType: TimeProviderType,
+      applicationId: ApplicationId,
+      parties: TriggerParties,
+  )(implicit loggingContext: LoggingContextOf[Trigger]): Runner = {
+    val setupId: UUID = UUID.randomUUID()
+
+    loggingContext.withEnrichedTriggerContext(
+      triggerAction("trigger.setup", id = setupId)
+    ) { implicit loggingContext: LoggingContextOf[Trigger] =>
+      loggingContext
+        .withEnrichedTriggerContext(
+          "level" -> trigger.defn.level.toString,
+          "version" -> trigger.defn.version.toString,
+        ) { implicit loggingContext: LoggingContextOf[Trigger] =>
+          new Runner(
+            compiledPackages,
+            trigger,
+            triggerConfig,
+            client,
+            timeProviderType,
+            applicationId,
+            parties,
+          )
+        }
+    }
+  }
+
+  def triggerAction(
+      name: String,
+      id: UUID = UUID.randomUUID(),
+      parent: Option[UUID] = None,
+  ): (String, LoggingValue) = {
+    "action" -> LoggingValue.Nested(
+      LoggingEntries(
+        "type" -> name,
+        "id" -> id,
+      ) ++ parent.fold(LoggingEntries.empty)(id => LoggingEntries("parentId" -> id))
+    )
+  }
```

Comment:
In future PRs, we'll use this method to define logging "spans" and so capture child/parent relationships. This will allow us to view/model trigger behaviour in process algebra like manner.